### DISCUSSION TITLE
Multi-enzyme spectral library question

### DISCUSSION THREAD

**Discussion by chrishuges** [vdemichev/DiaNN#581] [Q&A] [answered]
I am wondering about how best to create a spectral library using DIA-NN in a scenario where you have multiple peptides derived from different digests in a single sample. For example, you have digested a sample with trypsin and lysC independently, and then recombined the digests for analysis in a single DIA run. In a library-free analysis, I could see a few ways of going about this:

1. Turn up the missed cleavages and hope that the majority of your lysC peptides are represented in the trypsin generated library. This has some obvious issues. 
2. Generate a library for trypsin and search the files. Generate a second library for lysC and search the files. Combine the results downstream.
3. Concatenate the trypsin and lysC libraries eliminating redundant entries, and search the files against this combined library. I am not entirely sure at the moment what the best way to do this combination would be. 

Has anyone tried something like this?

Thanks,
Chris

**Comment by chrishuges** [comment]
I ended up solving this after reading [a nice paper](https://pubmed.ncbi.nlm.nih.gov/35234472/) that came out earlier this year that did something very similar: . Basically, I used the [Rapid Peptides Generator](https://rapid-peptide-generator.readthedocs.io/en/latest/index.html) to generate my peptide list that I reshaped into a suitable input for Prosit and used the output spectral library for DiaNN. May not be optimal as I would prefer to use the built-in DiaNN library functionality, but works for now.

**Reply by educhicano** [reply]
I remember that you can do Tryp/LysC enzymatic digestion by using --cut --cut K*,R* . But I'm not sure. The first --cut is to disable the default option and the second --cut is used to give DIANN the enzyme scheme. Can you try it?

**Comment by vdemichev** [comment, marked as answer]
Hi Chris,

Can do this purely in DIA-NN.
What DIA-NN can do:
- Generate a .predicted.speclib from a FASTA database with a particular digest specificity
- Save any spectral library to .tsv, including any .predicted.speclib library
- Load multiple .tsv libraries (with --lib commands) and use them / merge them into a single .tsv

So two options:
- Just use --cut like @educhicano indicated to combine all enzymes specificities and set up missed cleavages to a sufficiently high value
- Generate an in silico library separately for each enzyme, then convert to .tsv and combine

Best,
Vadim

**Reply by chrishuges** [reply]
Thank you @educhicano and @vdemichev for the replies, very helpful.

Is it necessary to use two --cut flags? I generated the libraries individually using just one --cut command (e.g. --cut R* for argC), and it seems OK. Does DIA-NN try to remove any redundant entries from the library, or does it just straight up combine them?

In case someone else stumbles upon this wondering the same thing, I ended up doing what Vadim suggested:

1. Create individual spectral libraries for trypsin (--cut K*,R*), lysC (--cut K*), argC (--cut R*) using DIA-NN.
2. Convert each .speclib file spectral library to .tsv using DIA-NN. 
3. Use multiple -llib commands to combine the individual .tsv libraries in DIA-NN. 
4. Use this combined library to search all of my files.
